### PR TITLE
Logan/2/issue3

### PR DESCRIPTION
**Pull request by loganfillo**

### CODE REVIEW

**Comment by kevinh42** [comment]
Might just be an issue on my end but I get the following error when I run `teleop_sim_bringup_launch.py`:
`[rviz2-8] [INFO] [1621120719.127591358] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-8] [INFO] [1621120719.127754678] [rviz2]: OpenGl version: 4.6 (GLSL 4.6)
[rviz2-8] [INFO] [1621120719.157610333] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-8] [INFO] [1621120719.318465659] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-8] [INFO] [1621120719.335418895] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-8] [INFO] [1621120719.353541672] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-8] [INFO] [1621120719.373201653] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-8] Parsing robot urdf xml string.
[ERROR] [rviz2-8]: process has died [pid 75313, exit code -11, cmd '/opt/ros/foxy/lib/rviz2/rviz2 -d /home/kevin/Git/triton/install/triton_gazebo/share/triton_gazebo/config/rviz_config.rviz --ros-args -r __node:=rviz2'].`

**Comment by JakeCronin1997** [comment]
> Might just be an issue on my end but I get the following error when I run `teleop_sim_bringup_launch.py`:
> `[rviz2-8] [INFO] [1621120719.127591358] [rviz2]: Stereo is NOT SUPPORTED [rviz2-8] [INFO] [1621120719.127754678] [rviz2]: OpenGl version: 4.6 (GLSL 4.6) [rviz2-8] [INFO] [1621120719.157610333] [rviz2]: Stereo is NOT SUPPORTED [rviz2-8] [INFO] [1621120719.318465659] [rviz2]: Stereo is NOT SUPPORTED [rviz2-8] [INFO] [1621120719.335418895] [rviz2]: Stereo is NOT SUPPORTED [rviz2-8] [INFO] [1621120719.353541672] [rviz2]: Stereo is NOT SUPPORTED [rviz2-8] [INFO] [1621120719.373201653] [rviz2]: Stereo is NOT SUPPORTED [rviz2-8] Parsing robot urdf xml string. [ERROR] [rviz2-8]: process has died [pid 75313, exit code -11, cmd '/opt/ros/foxy/lib/rviz2/rviz2 -d /home/kevin/Git/triton/install/triton_gazebo/share/triton_gazebo/config/rviz_config.rviz --ros-args -r __node:=rviz2'].`

I'm still having my own issues running Gazebo locally, I have a similar string of messages leading to the crash by the last one is different in my case.

File: src/triton_vision_utils/include/triton_vision_utils/contour_features.hpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+public:
+  ContourFeatures();
+
+  /**
+   * Produces ellipse features of the contour
+   * @param cnt A convex hull contour
+   * @return A vector containing the major asix (MA), minor axis (ma) and angle of
+   *           contour in that order
+   */
+  std::vector<float> ellispe_features(std::vector<cv::Point> cnt);
```
Comment (JakeCronin1997) [inline src/triton_vision_utils/include/triton_vision_utils/contour_features.hpp:23]: Should this be `ellipse_features`?

File: src/triton_vision_utils/include/triton_vision_utils/contour_features.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  /**
+   * Produces area features of the contour
+   * @param cnt A convex hull contour
+   * @return A vecotr containing the contour area, bounding rect area, aspect ratio.
```
Comment (JakeCronin1997) [inline src/triton_vision_utils/include/triton_vision_utils/contour_features.hpp:28]: vecotr->vector

**Review by JakeCronin1997 — COMMENTED** [review]
I noticed you're using 2-space indents on your source files in this PR, was this on purpose? if so we should create a ticket to make sure we're using the same style.

**Comment by kevinh42** [comment]
> I noticed you're using 2-space indents on your source files in this PR, was this on purpose? if so we should create a ticket to make sure we're using the same style.

Adding a .clang-format configuration file for the repo can help us keep the style consistent. Most IDEs can use it to auto-format code.

**Pull request merged**